compiler/rustc_session/src/filesearch.rs

```diff
@@ -113,6 +113,10 @@ pub fn make_target_lib_path(sysroot: &Path, target_triple: &str) -> PathBuf {
     sysroot.join(&relative_target_lib_path(sysroot, target_triple))
 }
 
+// This function checks if sysroot is found using env::args().next(), and if it
+// is not found, uses env::current_exe() to imply sysroot. However, since
+// env::args().next() can be set to an arbitrary value, this function and its
+// return value should not be relied on for security-related decisions.
```

Comment:
```suggestion
// is not found, uses env::current_exe() to imply sysroot.
```

A dynamically linked codegen backend can be loaded from the sysroot, so we have to rely on it for security-related decisions. This shouldn't be a problem though, as a program that can pass an arbitrary value for the first argument can likely run arbitrary code anyway.